src/components/filters/AttributeFilter/AttributeElements.tsx

```diff
@@ -0,0 +1,226 @@
+
+export interface IPaging {
+    count: number;
+    offset: number;
+    total: number;
+}
+
+export interface IAttributeElementsProps {
+    metadata?: typeof GoodData.md;
+    projectId: string;
+    uri?: string;
+    identifier?: string;
+    options?: GoodData.IValidElementsOptions;
+
+    children?(props: IAttributeElementsChildren): any;
+}
+
+export interface IValidElements {
+    items: GoodData.IElement[];
+    paging: IPaging;
+    elementsMeta: {
+        attribute: string;
+        attributeDisplayForm: string;
+        filter: string;
+        order: AFM.SortDirection;
+    };
+}
+
+export interface IAttributeElementsState {
+    validElements?: IValidElements;
+    isLoading: boolean;
+    error?: any;
+}
+
+export interface IAttributeElementsChildren {
+    validElements: IValidElements;
+    loadMore: Function;
+    isLoading: boolean;
+    error: any;
+}
+
+export class AttributeElements extends React.PureComponent<IAttributeElementsProps, IAttributeElementsState> {
+
+    public static propTypes = {
+        projectId: PropTypes.string.isRequired,
+        uri: PropTypes.string,
+        identifier: PropTypes.string,
+        options: PropTypes.object,
+        metadata: PropTypes.shape({
+            getObjectDetails: PropTypes.func.isRequired,
+            getObjectUri: PropTypes.func.isRequired
+        })
+    };
+
+    public static defaultProps: Partial<IAttributeElementsProps> = {
+        metadata: GoodData.md,
+        projectId: null,
+        uri: null,
+        identifier: null,
+        options: null,
+        children: ({
+            validElements,
+            loadMore,
+            isLoading,
+            error
+        }) => {
+            const paging: Partial<IPaging> = validElements ? validElements.paging : {};
+            const {
+                offset = 0,
+                count = null,
+                total = null
+            } = paging;
+            const nextOffset = count + offset;
+            if (error) {
+                return <div>{error}</div>;
+            }
+            return (
+                <div>
+                    <p>
+                        Use children function to map {'{'} validElements, loadMore, isLoading {'} '}
+                        to your React components.
+                    </p>
+                    <button
+                        onClick={loadMore as any}
+                        disabled={isLoading || (offset + count === total)}
+                    >More
+                    </button>
+                    <h2>validElements</h2>
+                    <pre>
+                        isLoading: {isLoading.toString()}
+                        offset: {offset}
+                        count: {count}
+                        total: {total}
+                        nextOffset: {nextOffset}
+                        validElements:
+                        {JSON.stringify(validElements, null, '\t')}
+                    </pre>
+                </div>
```

Comment:
Definitely remove "example children" from component`s implementation. Story is the right place - and there it is as well.